trl/trainer/utils.py

```diff
@@ -82,9 +82,14 @@ def __init__(
         **kwargs,
     ):
         super().__init__(*args, mlm=mlm, **kwargs)
+        self.ignore_index = ignore_index
         self.instruction_template = instruction_template
         self.response_template = response_template
-        self.ignore_index = ignore_index
+        if type(instruction_template) == list:
```

Comment:
```suggestion
        if isinstance(instruction_template, list):
```

Reply:
Thanks, that was fairly stupid of me to copy existing code like that, will also update the response template type check